ecommerce/app/controllers/customers_controller.rb

```diff
@@ -0,0 +1,71 @@
+class CustomersController < ApplicationController
+  before_action :set_customer, only: [:show, :edit, :update, :destroy]
+
+  # GET /customers
+  # GET /customers.json
+  def index
+    @customers = Customer.all
+  end
+
+  # GET /customers/new
+  def new
+    @customer = Customer.new
+  end
+
+  def orders
+    @customer = Customer.find(params[:id])
+    @orders = Order.where(customer_name: @customer.name)
+  end
+
+  # POST /customers
+  # POST /customers.json
+  def create
+    @customer = Customer.new(customer_params)
+    respond_to do |format|
+      if @customer.save
+        format.html { redirect_to @customer, notice: 'Customer was successfully created.' }
+        format.json { render :show, status: :created, location: @customer }
+      else
+        format.html { render :new }
+        format.json { render json: @customer.errors, status: :unprocessable_entity }
+      end
+    end
+   end
+
+  # PATCH/PUT /customers/1
+  # PATCH/PUT /customers/1.json
+  def update
+    respond_to do |format|
+      if @customer.update(customer_params)
+        format.html { redirect_to @customer, notice: 'Customer was successfully updated.' }
+        format.json { render :show, status: :ok, location: @customer }
+      else
+        format.html { render :edit }
+        format.json { render json: @customer.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # DELETE /customers/1
+  # DELETE /customers/1.json
+  def destroy
+    set_customer
+    Order.where(customer_name: @customer.name.to_s).destroy_all
```

Comment:
misuse of name here, never use string values to update or delete, use id instead.
Use single query for lines 52 and 53, use joins